### PR TITLE
Feature/add group resource

### PR DESCRIPTION
**Pull request by maikelpoot**
This PR will enable the provider to create groups.

Update should trigger warning and do nothing
Delete should remove the resource from the terraform state, but won't delete from keyhub

Usage:
```hcl
#Create a group
resource "keyhub_group" "new_group" {
  name = "new_keyhub_group"

  # UUID of first keyhub user/account
  member {
    uuid = "00000000-0000-0000-0000-000000000000"
    rights = "MANAGER"
  }

  # UUID of second keyhub user/account
  member {
    uuid = "11111111-1111-1111-1111-111111111111"
    rights = "MANAGER"
  }

  audit_months = ["JANUARY"]

  description = "This group is created by terraform"

  # UUID of the group that should be set as authoring group
  provisioning_auth_groupuuid = "00000000-0000-0000-0000-000000000000"
  membership_auth_groupuuid   = "00000000-0000-0000-0000-000000000000"
  auditing_auth_groupuuid     = "00000000-0000-0000-0000-000000000000"

}
```

### CODE REVIEW

File: keyhub/data_source_provisionedsystem.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	UserDN                     string       `json:"userDN,omitempty"`
+	GroupDN                    string       `json:"groupDN,omitempty"`
+	SshPublicKeySupported      bool         `json:"sshPublicKeySupported,omitempty"`
+	ObjectClasses              string       `json:"objectClasses,omitempty"`
+	TrustedCertificate         *Certificate `json:"trustedCertificate,omitempty"`
+	FailoverTrustedCertificate *Certificate `json:"failoverTrustedCertificate,omitempty"`
+	ClientCertificate          *Certificate `json:"clientCertificate,omitempty"`
+	Attributes                 interface{}  `json:"attributes,omitempty"`
+}
+*/
```
Comment (hielkehoeve) [inline keyhub/data_source_provisionedsystem.go:37]: Is this still necessary?

**Comment by hielkehoeve** [comment]
You are using `computed` on nearly all properties, while `computed` has a different meaning:
https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#computed

File: keyhub/data_source_provisionedsystem.go (hunk 1 above)
Comment (maikelpoot) [reply keyhub/data_source_provisionedsystem.go]: Removed

**Comment by maikelpoot** [comment]
> You are using `computed` on nearly all properties of data_source_group, while `computed` has a different meaning: https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#computed

> Computed is often used to represent values that are not user configurable or can not be known at time of terraform plan or apply

in case of data_source_group (or most other datasources), UUID is the only field that the user should configur. Al the other values are retrieved from the backend, and are, in terms of terraform, computed.

If you omit computed, you get the error below:
```
 Error while loading schemas for plugin components: Failed to obtain provider schema: Could not load the schema for provider
│ registry.terraform.io/topicuskeyhub/keyhub: provider registry.terraform.io/topicuskeyhub/keyhub has invalid schema for data resource type
│ "keyhub_group", which is a bug in the provider: "1 error occurred:\n\t* vault_recovery: must set Optional, Required or Computed\n\n"..
```
As the fields (for the user) are not optional (it should not be filled by the user), or required, computed is the only option.

**Comment by hielkehoeve** [comment]
My bad, at time of writing I thought you also used the schema for resource. Should have payed more attention  🤪

**Review by hielkehoeve — APPROVED** [review]

**Pull request merged**